assets/js/timetree.js

```diff
@@ -8,119 +8,136 @@ import {schemeGreens} from "d3-scale-chromatic"
 const d3 = {select, forceSimulation, forceManyBody, forceCenter, forceCollide,
     forceLink, forceY, line, curveNatural, scaleSequential, schemeGreens};
 
+// load & parse leaf data from json embedded in the document
+const leafData = document.querySelector('.leaf-data');
+const data = JSON.parse(leafData.value);
+
+// generate list of centuries referenced in the data; sort most recent first
+let centuries = Array.from(new Set(
+  data.leaves.filter(leaf => leaf.century != undefined)
+      .map(leaf => leaf.century))
+  ).sort().reverse();
+
+// check and report on total leaves, unsortable leaves
+console.log(`${data.leaves.length} total leaves`);
+// NOTE: some sort dates set to empty string, "?"; 0 is allowed for earliest sort
+let unsortableLeaves = data.leaves.filter(leaf => leaf.sort_date === null || leaf.sort_date == "");
+console.log(`${unsortableLeaves.length} lea${unsortableLeaves.length == 1 ? "f" : "ves"} with sort date not set`);
+
+// ignore any records with sort date unset
+let sortedLeaves = data.leaves.filter(leaf => leaf.sort_date != null)
+  .sort((a, b) => a.sort_date > b.sort_date)
+// use url as id for node in graph; set type to leaf; set century
+sortedLeaves.forEach(leaf => {
+  leaf.id = leaf.url;
+  leaf.type = "leaf";
+  // set place-holder centuries for special cases
+  // (century set in json based on sort date if numeric)
+  if (leaf.century == undefined) {
+    if (leaf.sort_date == 0 || leaf.sort_date == "") {
+      // put zeros in the 1500s
+      leaf.century = 15;
+    } else if (leaf.sort_date == "TBA" || leaf.sort_date == "?") {
+      // put TBs / ? in the 2000s
+      leaf.century = 20;
+    }
+  }
+});
 
-fetch('index.json')
-  .then((response) => response.json())
-  .then((data) => {
```

Comment:
Nice. I prefer the new solution where all of this isn't embedded in the fetch function